zipline/pipeline/loaders/blaze.py

```diff
@@ -0,0 +1,697 @@
+    """Inline any deltas into the base set that would have changed our most
+    recently known value.
+
+    Parameters
+    ----------
+    base : pd.DataFrame
+        The first known values.
+    deltas : pd.DataFrame
+        Overwrites to the base data.
+    dates : pd.DatetimeIndex
+        The dates requested by the loader.
+
+    Returns
+    -------
+    new_base : pd.DataFrame
+        The new base data with novel deltas inserted.
+    """
+    get_indexes = dates.searchsorted
+    return pd.concat(
+        (base,
+         deltas.loc[
+             (get_indexes(deltas[TS_FIELD_NAME].values, 'right') -
+              get_indexes(deltas[AD_FIELD_NAME].values, 'letf')) <= 1
+         ].drop(AD_FIELD_NAME, 1)),
+        ignore_index=True,
+    )
+
+
+def overwrite_from_dates(asof, dates, dense_dates, asset_idx, value):
+    """Construct a `Float64Overwrite` with the correct
+    start and end date based on the asof date of the delta,
+    the dense_dates, and the dense_dates.
+
+    Parameters
+    ----------
+    asof : datetime
+        The asof date of the delta.
+    dates : pd.DatetimeIndex
+        The dates requested by the loader.
+    dense_dates : pd.DatetimeIndex
+        The dates that appeared in the dataset.
+    asset_idx : int
+        The index of the asset in the block.
+    value : np.float64
+        The value to overwrite with.
+
+    Returns
+    -------
+    overwrite : Float64Overwrite
+        The overwrite that will apply the new value to the data.
+    """
+    return Float64Overwrite(
+        dates.searchsorted(asof),
+        dates.get_loc(dense_dates[dense_dates.searchsorted(asof) + 1]) - 1,
+        asset_idx,
+        value,
+    )
+
+
+def adjustments_from_deltas_no_sids(dates,
+                                    dense_dates,
+                                    column_idx,
+                                    column_name,
+                                    assets,
+                                    deltas):
+    """Collect all the adjustments that occur in a dataset that does not
+    have a sid column.
+
+    Parameters
+    ----------
+    dates : pd.DatetimeIndex
+        The dates requested by the loader.
+    dense_dates : pd.DatetimeIndex
+        The dates that were in the dense data.
+    column_idx : int
+        The index of the column in the dataset.
+    column_name : str
+        The name of the column to compute deltas for.
+    deltas : pd.DataFrame
+        The overwrites that should be applied to the dataset.
+
+    Returns
+    -------
+    adjustments : dict[idx -> Float64Overwrite]
+        The adjustments dictionary to feed to the adjusted array.
+    """
+    return {
+        dates.get_loc(kd): tuple(
+            overwrite_from_dates(
+                deltas.loc[kd, AD_FIELD_NAME],
+                dates,
+                dense_dates,
+                n,
+                v,
+            ) for n in range(len(assets))
+        ) for kd, v in deltas[column_name].iteritems()
```

Comment:
it looks it up once and then leave the iter on the stack. I did refactor the `deltas` above to cache the `deltas.loc[: AD_FIELD_NAME]`